src/System.Management.Automation/engine/runtime/Binding/Binders.cs

```diff
@@ -4258,13 +4266,16 @@ private DynamicMetaObject GetIndexMultiDimensionArray(DynamicMetaObject target,
                     target.CombineRestrictions(indexes));
             }
 
-            if (getterParams.Length == 1 && getterParams[0].ParameterType == typeof(int) && CanIndexFromEndWithNegativeIndex(target))
+            if (CanIndexFromEndWithNegativeIndex(target, getter, getterParams))
             {
                 // PowerShell supports negative indexing for some types (specifically, types implementing IList or IList<T>).
                 // For those types, generate special code to check for negative indices, otherwise just generate
                 // the call.
-                PropertyInfo lengthProperty = target.LimitType.GetProperty("Count") ??
-                                              target.LimitType.GetProperty("Length"); // for string
+                if (lengthProperty == null)
+                {
+                    lengthProperty = target.LimitType.GetProperty("Count") ??
+                                     target.LimitType.GetProperty("Length"); // for string
```

Comment:
Comment should be on separate lines. Please remove the comment or add enhanced comment above.

Reply:
Removed the inline comment and added a more description comment above.